tools/codegen/api/dispatcher.py

```diff
@@ -74,18 +74,23 @@ def arguments(func: FunctionSchema) -> Sequence[DispatcherArgument]:
 
 # Given a set of CppArguments in scope, return a sequence of dispatcher
 # expressions that translate the cpp API into dispatcher API
-def cppargument_exprs(a: CppArgument, *, tensor_options: Optional[CppArgument]) -> Sequence[DispatcherExpr]:
+def cppargument_exprs(a: CppArgument, *, tensor_options: Optional[CppArgument], process_tensoroptions: Optional[str]) -> Sequence[DispatcherExpr]:
```

Comment:
Not sure if I agree with the general API design here, but at the very least, `process_tensoroptions` should be an Enum, not a freeform string.

Reply:
I think just an optional bool `gather` should do it. (gather rather than scatter because gather is the special case that's going away.)

Reply:
I don't think an optional bool would cut it here. It's hard to understand that `gather=None` means "do nothing" but `gather=False` means "scatter"...I made it an enum